Observer/SpeakFasterObserver/AudioAsr.cs

```
@@ -136,10 +208,122 @@ private void ReInitStreamRecognizer()
+            int bufferEndIndex = bytesPerSample * (int)(audioFormat.SampleRate * endTime);
+            byte[] snippetBuffer = new byte[bufferEndIndex - bufferStartIndex];
+            lock (speakerIdBufferLock)
+            {
+                Array.Copy(
+                    speakerIdBuffer, bufferStartIndex, snippetBuffer, 0,
+                    bufferEndIndex - bufferStartIndex);
+            }
+            SendSpeakerIdHttpRequest(snippetBuffer);
+        }
+
+        private async void SendSpeakerIdHttpRequest(byte[] snippetBuffer)
+        {
+            if (speakerIdEndpoint == null || speakerIdSubscriptionKey == null ||
+                speakerMap.Count == 0)
+            {
+                return;
+            }
+            using (var client = new HttpClient())
+            {
+                string[] profileIds = new string[speakerMap.Count];
+                speakerMap.Keys.CopyTo(profileIds, 0);
+                string url =
+                    $"{speakerIdEndpoint}/speaker/identification/v2.0/text-independent/profiles/" +
+                    $"identifySingleSpeaker?profileIds={string.Join(",", profileIds)}";
+                ByteArrayContent byteArrayContent = new(snippetBuffer);
+                byteArrayContent.Headers.Add(
+                    "ContentType",
+                    $"audio/wav; codecs=audio/pcm; samplerate={audioFormat.SampleRate}");
+                byteArrayContent.Headers.Add("Ocp-Apim-Subscription-Key", speakerIdSubscriptionKey);
+                var response = await client.PostAsync(url, byteArrayContent);
+                PrintDetectedSpeakerName(response);
+            }
+        }
+
+        private async void PrintDetectedSpeakerName(HttpResponseMessage speakerIdResponse)
+        {
+            if (speakerIdResponse.StatusCode != HttpStatusCode.OK)
+            {
+                string errorString = await speakerIdResponse.Content.ReadAsStringAsync();
+                Debug.WriteLine(
```

Typically don't like to see Debug.WriteLine in production code, as this ends up hiding errors when the code is run with no debugger attached.

My preference is to 'fail loud', e.g. throw a messagebox and exit, as I've seen lots of hard to uncover 'silently failed 30 function calls ago' in my time...